src/components/interactive-map.js

```diff
@@ -121,102 +168,125 @@ export default class InteractiveMap extends PureComponent {
   }
 
   _handleEvent(event) {
-    const {mapControls} = this.props;
-    // MapControls only extracts the states that it recognizes.
-    // This allows custom map controls to add new states and callbacks.
-    mapControls.setState(Object.assign({}, this.props, {
-      mapState: new MapState(this.props)
-    }));
-
-    return mapControls.handle(event);
+    const controlOptions = Object.assign({
```

Comment:
Hm, was this done specifically so that `onChangeState` could be overridden from props?